datafusion/core/tests/parquet/arrow_statistics.rs

```diff
@@ -0,0 +1,528 @@
+    let rg_statistics = parquet_row_group_statistics(0, 4, 7, row_per_group);
+
+    //////////// Verify write row group statistics ////////////
+    let write_rg_stats = &rg_statistics.write_statistics;
+
+    // only 1 row group
+    assert_eq!(write_rg_stats.len(), 1, "expected 1 row group");
+    let rg = &write_rg_stats[0];
+
+    // 3 rows
+    assert_eq!(rg.row_count, 3, "expected 3 rows");
+
+    // no nulls
+    assert!(rg.statistics.null_count.is_none(), "expected no nulls");
+
+    let statistics = &rg.statistics;
+
+    // min is [4, 0, 0, 0, 0, 0, 0, 0]
+    assert_eq!(
+        *statistics.min_value.as_ref().unwrap(),
+        [4, 0, 0, 0, 0, 0, 0, 0]
+    );
+    // max is [6, 0, 0, 0, 0, 0, 0, 0]
+    assert_eq!(
+        *statistics.max_value.as_ref().unwrap(),
+        [6, 0, 0, 0, 0, 0, 0, 0]
+    );
+
+    println!("RG statistics:\n {statistics:#?}");
+    // This will display:
+    // RG statistics:
+    // Statistics {
+    //     max: Some(
+    //         [
+    //             6,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //         ],
+    //     ),
+    //     min: Some(
+    //         [
+    //             4,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //         ],
+    //     ),
+    //     null_count: None,
+    //     distinct_count: None,
+    //     max_value: Some(
+    //         [
+    //             6,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //         ],
+    //     ),
+    //     min_value: Some(
+    //         [
+    //             4,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //             0,
+    //         ],
+    //     ),
+    //     is_max_value_exact: Some(
+    //         true,
+    //     ),
+    //     is_min_value_exact: Some(
+    //         true,
+    //     ),
+    // }
+
+    //////////// Verify read row group statistics ////////////
+    let read_rg_stats = &rg_statistics.read_statistics;
+
+    // only 1 row group
+    assert_eq!(read_rg_stats.len(), 1, "expected 1 row group");
+    let rg = &read_rg_stats[0];
+
+    // 3 rows
+    assert_eq!(rg.row_count, 3, "expected 3 rows");
+
+    // rg.statistics is Int64
+    let stats = &rg.statistics;
+    match stats {
+        ParquetFileStatistics::Int64(int64_stats) => {
+            // no nulls
+            assert_eq!(int64_stats.null_count(), 0, "expected no nulls");
+            // min is 4
+            assert_eq!(*int64_stats.min(), 4);
+            // max is 6
+            assert_eq!(*int64_stats.max(), 6);
+        }
+        _ => panic!("expected Int64"),
+    }
```

Comment:
Rather than verifying the data that came out of the parquet reader, ideally here we would call `parquet_stats_to_arrow` and verify the output

For example, something like

```rust
// figure out the arrow datatype for "i64" column
let data_type = get_datatype_for_column("i64");
let stats = parquet_stats_to_arrow(
  data_type,
  rg.statistics.iter()
);

// verify that the min/max came out right
assert_eq!(stats.min, ArrayRef::new(Int32Array::from(vec![4, 0, 0]))));
// same for max
```

Reply:
Yes, that is the plan. These tests are mostly for me to understand what the parquet stats looks like before I convert it into the arrow stats

Reply:
Makes sense -- sorry this wasn't as ready to work on as I may have hoped